code-challenges.test.js

```diff
@@ -13,40 +13,176 @@
+    //return the array with strings that only has the letter
+//test function
+
+// const finder = (array, letter) => {
+    
+//    return array.filter(value=> value.includes(letter))
+// }
+
+// - Expected  - 1
+// + Received  + 0
+
+const finder = (array, letter) => {
+    let newArray = array.map(value=>value.toLowerCase())
+    return newArray.filter(value=> value.includes(letter)).map(value=>value.charAt(0).toUpperCase() + value.slice(1))
+}
+
+// Test Suites: 1 passed, 1 total
+// Tests:       2 passed, 2 total
+
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
-// Expected output: true
+
+describe("fullHouse", () => {
+    it("takes in an array of 5 numbers and determines whether or not the array is a full house", () => {
+        const hand1 = [5, 5, 5, 3, 3]
+        let output6 = true
+        const hand2 = [5, 5, 3, 3, 4]
+        let output7 = false
+        const hand3 = [5, 5, 5, 5, 4]
+        let output8 = false
+        const hand4 = [7, 2, 7, 2, 7]
+        let output9 = true
+      expect(fullHouse(hand1)).toEqual(output6)
+      expect(fullHouse(hand2)).toEqual(output7)
+      expect(fullHouse(hand3)).toEqual(output8)
+      expect(fullHouse(hand4)).toEqual(output9)
+    })
+  })
+
+//   ReferenceError: fullHouse is not defined
 
 // b) Create the function that makes the test pass.
+
+//create a function called fullHouse
+//it will take in the parameter of an array
+//create an object to keep count
+  //create a for loop where it looks at the value at each index of the array
+  //the object will keep count for each value found in the for loop starting at 0 and adding 1 for each one found
+  //create a variable to check the object count of each value
+  // an if statement will check if the values of the object will pass as a full house by checking whether each one is equal to a count of 3 and 2.
+  //if it does then it returns true
+  //if not then it returns false
+//test the function
+
+// const fullHouse = (array) => {
+    
+//     for(i=0;i<array.length;i++){
+//         for(j=0;j<array.length;j++)
+//         if(array.indexOf(array[i], 3) && array.indexOf(array[j],2)){
+//             return true
+//         }
+//         else {return false}
+//     }
+// }
+// Expected: false
+// Received: true
+// const fullHouse = (array) => {
+//   let check = array.slice(0,1)
+//   for(i=0;i<array.length;i++)
+//     if(array[i]===check)
+//         {return true}
+//     else {return false}
+// }
+
+// Expected: true
+// Received: false
+
+const fullHouse = (array) => {
```

Comment:
Excellent work on this problem!  Really like this approach and your pseudocode was excellent at explaining what is happening.